updateQDKVersion.ps1

```diff
@@ -0,0 +1,61 @@
+$ver = $args[0]
+Write-Host $ver
+
+$csString = 'PackageReference Include=\"Microsoft\.Quantum\.[a-zA-Z\.]+\" Version=\"(?<oldVersion>[0-9|\.]+)\"'
+$csFiles = (Select-String -Path "**\*.csproj" -pattern "Microsoft.Quantum" | Select-Object -Unique Path)
+$csFiles | ForEach-Object {
+    (Get-Content -Encoding UTF8 $_.Path) | ForEach-Object {
+         $isQuantumPackage = $_ -match $csString
+         if ($isQuantumPackage) {
+             $_ -replace $Matches.oldVersion, $ver
+         } else {
+             $_
+         }
+    } | Set-Content -Encoding UTF8 $_.Path
+}
+
+$ipynb_string = '%package Microsoft.Quantum.Katas::(?<oldVersion>[0-9|\.]+)'
+$ipynbFiles =  (Select-String -Path "**\*.ipynb" -pattern "Microsoft.Quantum" | Select-Object -Unique Path)
+
+$ipynbFiles | ForEach-Object {
+    if ($_)
+    {
+        (Get-Content $_.Path) | ForEach-Object {
+            $isQuantumPackage = $_ -match $ipynb_string
+            if ($isQuantumPackage) {
+                $_ -replace $Matches.oldVersion, $ver
+            } else {
+                $_
+            }
+        } | Set-Content $_.Path
+    }
+}
+
+$docker_string = 'FROM mcr.microsoft.com/quantum/iqsharp-base:(?<oldVersion>[0-9|\.]+)'
+$dockerFiles =  (Select-String -Path "Dockerfile" -pattern "microsoft.com/quantum" | Select-Object -Unique Path)
+$dockerFiles | ForEach-Object {
+    (Get-Content $_.Path) | ForEach-Object {
+         $isQuantumPackage = $_ -match $docker_string
+         if ($isQuantumPackage) {
+             $_ -replace $Matches.oldVersion, $ver
+         } else {
+             $_
+         }
+    } | Set-Content $_.Path
+}
+
+$ps1_string = 'Microsoft.Quantum.IQSharp --version (?<oldVersion>[0-9|\.]+)'
+$ps1Files = (Select-String -Path "**\install-iqsharp.ps1" -pattern "Microsoft.Quantum.IQSharp" | Select-Object -Unique Path)
+$ps1Files | ForEach-Object {
+    
+    (Get-Content -Encoding 'Windows-1252' $_.Path) | ForEach-Object {
```

Comment:
My PowerShell doesn't recognize encoding Windows-1252:

    Cannot bind parameter 'Encoding'. Cannot convert value "Windows-1252" to type
    "Microsoft.PowerShell.Commands.FileSystemCmdletProviderEncoding". Error: "Unable to match the identifier name
    Windows-1252 to a valid enumerator name. Specify one of the following enumerator names and try again:
    Unknown, String, Unicode, Byte, BigEndianUnicode, UTF8, UTF7, UTF32, Ascii, Default, Oem, BigEndianUTF32"

I tried just omitting the encoding here and it worked just fine. Do you get any problems on your system if you omit the encoding?

Reply:
I'm with @tcNickolas on this; we should likely avoid Windows-1252 if possible, especially to help compatibility with non-Windows users.

Reply:
Nope. What encoding would you suggest @cgranade?

Reply:
@tcNickolas just noting the whole quantum operators here again. I changed it to jupyter notebook's native encoding, but that changed the quantum operators.

Reply:
Let's omit the encoding parameter in the commands for the non-csproj files if omitting the encoding allows to preserve them unchanged.